app/views/common/BackboneModalHarness.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+</template>
+
+<script>
+  /**
+   * Note a known limitation of this harness is that it does not support swapping out the following props
+   * after initial render:
+   *   - modalView
+   *   - modalOptions
+   *   - modalArgs
+   *
+   * In order to change these you must unmount and remount the component.
+   */
+  export default {
+    inject: ['openLegacyModal', 'legacyModalClosed'],
+
+    props: {
+      modalView: Function,
+
+      modalOptions: {
+        type: Object,
+        default: () => ({}),
+      },
+
+      modalArgs: {
+        type: Array,
+        default: () => ([]),
+      },
+
+      open: {
+        type: Boolean,
+        default: true
+      }
+    },
+
+    data: function () {
+      return {
+        modalLoading: false,
+        modalLoadingProgress: 0,
+
+        modalViewInstance: undefined
+      }
+    },
+
+    methods: {
+      loadModal () {
+        if (this.modalViewInstance) {
+          return
+        }
+
+        this.modalViewInstance = new this.$props.modalView(
+          this.$props.modalOptions,
+          ...this.$props.modalArgs,
+        )
+
+        this.modalViewInstance.on('loading:show', this.showLoadingEvent)
+        this.modalViewInstance.on('loading:hide', this.hideLoadingEvent)
+        this.modalViewInstance.on('loading:progress', this.updateLoadingProgressEvent)
+        this.modalViewInstance.on('hide', this.modalHideEvent)
+        this.modalViewInstance.on('hidden', this.modalHideEvent)
+      },
+
+      openModal () {
+        this.loadModal()
+        this.openLegacyModal(this.modalViewInstance)
+      },
+
+      closeModal() {
+        if (!this.modalViewInstance) {
+          return
+        }
+
+        this.modalViewInstance.destroy()
+        this.cleanupModal()
```

Comment:
A naive question, but does `destroy` keep a reference to something that can have events removed?
I sort of expect that events would get removed first. Then `destroy` called, and finally variable made undefined.

Reply:
Yeah it does but I like your thinking I'll rearrange.